dynaconf/validator_conditions.py

```diff
@@ -42,6 +42,13 @@ def identity(value, other):
 
 def is_type_of(value, other):
     """Type check"""
+
+    # auto transform quoted types
+    if isinstance(other, str):
+        from dynaconf.utils.parse_conf import type_map
+
+        other = type_map[other]
```

Comment:
```suggestion
        other = getattr(__builtins__, other)
```
this should work for all builtin types like str, int, dict, set etc

Reply:
Will do that

Reply:
For some reason, this raises an error in the app context (even though it works on the interpreter). I've assured `other` is a string with value of 'int', so this is really weird.

```
    other = getattr(__builtins__, other)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'dict' object has no attribute 'int'
```

But this works:

```python
other = __builtins__.get(other)
```